src/main/java/com/exasol/sql/expression/literal/BooleanLiteral.java

```diff
@@ -94,7 +87,12 @@ public void dismissConcrete(final BooleanExpressionVisitor visitor) {
     }
 
     @Override
-    public void accept(final ValueExpressionVisitor visitor) {
+    public void accept(final LiteralVisitor visitor) {
```

Comment:
This is interesting. Why does it accept two different visitors? This could indicate that the class is not used consistently in the tree that represents the SQL statement. Please check.